crates/pathfinder/src/storage/schema/revision_0011.rs

```diff
@@ -21,37 +21,136 @@ pub(crate) fn migrate(transaction: &Transaction) -> anyhow::Result<PostMigration
+    let mut parsing_time = std::time::Duration::default();
+
     while let Some(r) = rows.next()? {
         let transaction_hash = r.get_ref_unwrap("hash").as_blob()?;
         let receipt = r.get_ref_unwrap("receipt").as_blob()?;
+        let decompression_started = std::time::Instant::now();
         let receipt = zstd::decode_all(receipt).context("Decompress receipt")?;
-        let receipt: transaction::Receipt =
+        decompression_time += decompression_started.elapsed();
+
+        let parsing_started = std::time::Instant::now();
+        let receipt: LightReceipt =
             serde_json::de::from_slice(&receipt).context("Deserializing transaction receipt")?;
+        parsing_time += parsing_started.elapsed();
 
         receipt.events.into_iter().enumerate().try_for_each(
             |(idx, event)| -> anyhow::Result<_> {
-                transaction
-                    .execute(
-                        r"UPDATE starknet_events 
+                let mut stmt = transaction.prepare_cached(
+                    r"UPDATE starknet_events
                             SET from_address=:from_address
                             WHERE idx=:idx AND transaction_hash=:transaction_hash",
-                        named_params![
-                            ":idx": idx,
-                            ":transaction_hash": transaction_hash,
-                            ":from_address": &event.from_address.0.as_be_bytes()[..],
-                        ],
-                    )
-                    .context("Insert event data into events table")?;
+                )?;
+
+                stmt.execute(named_params![
+                    ":idx": idx,
+                    ":transaction_hash": transaction_hash,
+                    ":from_address": &event.from_address.0.as_be_bytes()[..],
+                ])
+                .context("Insert event data into events table")?;
 
                 Ok(())
             },
         )?;
+
+        processed_rows += 1;
+
+        if batch_size > 0 && processed_rows % batch_size == 0 {
+            let now = std::time::Instant::now();
+            let total_elapsed = now - start_of_run;
+            let batch_elapsed = now - start_of_batch;
+
+            let total_per_row = total_elapsed.div_f64(processed_rows as f64);
+            let batch_per_row = batch_elapsed.div_f64(batch_size as f64);
+
+            // this is non-scientific, but perhaps the latest helps? seems to be very much off until 75% when divided by 2, 50% when divided by 1.5
+            let est_per_row = (total_per_row + batch_per_row).div_f64(1.5);
+            let remaining = est_per_row * ((todo - processed_rows) as u32);
+
+            println!(
+                "{} rows processed, {:?} elapsed, {:?} per row in batch of {batch_size}, {remaining:?} remaining",
+                processed_rows,
+                total_elapsed,
+                batch_elapsed.div_f64(batch_size as f64),
+            );
```

Comment:
Should we make this a `tracing`? Same with the other `println`.

Reply:
Done this as well.